### DISCUSSION TITLE
Has error reporting changed?

### DISCUSSION THREAD

**Discussion by danielo515** [L3MON4D3/LuaSnip#660] [General]
Hello.
I think it is very nice that luasnips validates the snippets when you save the file.
Previously, if I forgot to "consume" a placeholder in a fmt string, I got an error telling me there were one unused placeholder (happens to me more commonly than I want to admit).
However, recently I started to get errors that tell me nothing about why they are failing. Has this changed? Or is it something related to how my neovim instance shows me the errors?

**Comment by L3MON4D3** [comment]
You mean lua-loaded snippets, right?
The error messages were changed to say something like "loading the file failed: <error>", but the original error should still be propagated (through <error>). Does this not happen?

**Comment by danielo515** [comment]
Yes, sorry if I was not specific enough. I mean the automatic reload of Lua
snippets when you save the file. I used to get a concrete error telling me
that one placeholder was not con consumed, and that is no longer the case.
Also I have noticed the autocommand is setup for *, why not limit it to Lua
files?

**Reply by L3MON4D3** [reply]
Mhmmm, I'm still getting the full error.
![screen](https://user-images.githubusercontent.com/41961280/202853661-e30653a2-c142-4a66-9cd9-35141fee9ec6.png)

I guess something else is up, what does your error look like?
Regarding the autocommand, (I'm assuming [this one](https://github.com/L3MON4D3/LuaSnip/blob/61238b9e0efb315ceebe579dbcb6729a6b602e9f/lua/luasnip/loaders/init.lua#L108-L110)), it handles snipmate, vscode, and lua-files. And then it's just easier to do that check whenever a file is saved, than doing it separately for all.

**Reply by danielo515** [reply]
![image](https://user-images.githubusercontent.com/2270425/202889046-f577515e-3009-4d29-bc01-43e79284597d.png)

This is how my error looks like. Very similar to yours, but with subtle details are different, like the line numbers it references (mine is 43 while yours is 63). The only thing missing is the `fmt.lua:88` message...

**Reply by L3MON4D3** [reply]
Ahh, you're on an older version of luasnip. There was a period where we had this shorter error, but it's reinstated now.
(not released though, you'd have to check out master for a bit)

**Comment by danielo515** [comment]
So, just to be clear, this was previously present (because I saw it on the
past) then removed (my current version) and then readded (latest version).
Is that right?

**Reply by L3MON4D3** [reply]
Yes: just pass the error through -> specify that the error occured while trying to load one of these files (omitting the specific error) -> also printing the error
